pkg/distributor/distributor_test.go

```diff
@@ -996,6 +996,56 @@ func TestDistributor_QueryStream_ShouldReturnErrorIfMaxSeriesPerQueryLimitIsReac
 	assert.Contains(t, err.Error(), "max number of series limit")
 }
 
+func TestDistributor_QueryStream_ShouldReturnErrorIfMaxChunkBytesPerQueryLimitIsReached(t *testing.T) {
+	const maxBytesLimit = 100
+
+	limits := &validation.Limits{}
+	flagext.DefaultValues(limits)
+	ctx = limiter.AddQueryLimiterToContext(ctx, limiter.NewQueryLimiter(0, maxBytesLimit))
+	// Prepare distributors.
+	ds, _, r, _ := prepare(t, prepConfig{
+		numIngesters:     3,
+		happyIngesters:   3,
+		numDistributors:  1,
+		shardByAllLabels: true,
+		limits:           limits,
+	})
+	defer stopAll(ds, r)
+
+	// Push a number of series below the max series limit.
+	initialSeries := maxBytesLimit / 60
```

Comment:
How was 60 computed? I think it's safer if you do something like this:

1. Call `writeReq := makeWriteRequest(0, 10, 0)` to generate a request with 10 series
2. Calculate the chunks size, multiply by replication factor, add +1: that's the limit
3. Query it back: should succeed
4. Add 1 more series
5. Query it back: should fail